This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Boris Pek <tehnick-8@yandex.ru>
Date: Wed, 9 Jan 2019 15:57:45 +0300
Subject: [PATCH 1/1] Fix build as static library for MS Windows using MinGW.

+ Delete extra definitions.

diff --git a/include/fcgi_stdio.h b/include/fcgi_stdio.h
index 1111111..2222222 100644
--- a/include/fcgi_stdio.h
+++ b/include/fcgi_stdio.h
@@ -24,7 +24,7 @@ extern "C" {
 #endif
 
 #ifndef DLLAPI
-#ifdef _WIN32
+#if defined (_WIN32) && defined (_MSC_VER)
 #define DLLAPI __declspec(dllimport)
 #else
 #define DLLAPI
diff --git a/include/fcgiapp.h b/include/fcgiapp.h
index 1111111..2222222 100644
--- a/include/fcgiapp.h
+++ b/include/fcgiapp.h
@@ -23,7 +23,7 @@
 #endif
 
 #ifndef DLLAPI
-#ifdef _WIN32
+#if defined (_WIN32) && defined (_MSC_VER)
 #define DLLAPI __declspec(dllimport)
 #else
 #define DLLAPI
diff --git a/include/fcgio.h b/include/fcgio.h
index 1111111..2222222 100644
--- a/include/fcgio.h
+++ b/include/fcgio.h
@@ -35,7 +35,7 @@
 #include "fcgiapp.h"
 
 #ifndef DLLAPI
-#ifdef _WIN32
+#if defined (_WIN32) && defined (_MSC_VER)
 #define DLLAPI __declspec(dllimport)
 #else
 #define DLLAPI
diff --git a/include/fcgios.h b/include/fcgios.h
index 1111111..2222222 100755
--- a/include/fcgios.h
+++ b/include/fcgios.h
@@ -50,7 +50,7 @@ extern "C" {
 #endif /* !_WIN32 */
 
 #ifndef DLLAPI
-#ifdef _WIN32
+#if defined (_WIN32) && defined (_MSC_VER)
 #define DLLAPI __declspec(dllimport)
 #else
 #define DLLAPI
diff --git a/libfcgi/fcgi_stdio.c b/libfcgi/fcgi_stdio.c
index 1111111..2222222 100644
--- a/libfcgi/fcgi_stdio.c
+++ b/libfcgi/fcgi_stdio.c
@@ -22,10 +22,6 @@
 #include <unistd.h>
 #endif
 
-#ifdef _WIN32
-#define DLLAPI  __declspec(dllexport)
-#endif
-
 #include "fcgiapp.h"
 #include "fcgios.h"
 #include "fcgimisc.h"
diff --git a/libfcgi/fcgiapp.c b/libfcgi/fcgiapp.c
index 1111111..2222222 100644
--- a/libfcgi/fcgiapp.c
+++ b/libfcgi/fcgiapp.c
@@ -40,10 +40,6 @@
 #include <limits.h>
 #endif
 
-#ifdef _WIN32
-#define DLLAPI  __declspec(dllexport)
-#endif
-
 #include "fcgimisc.h"
 #include "fastcgi.h"
 #include "fcgios.h"
diff --git a/libfcgi/fcgio.cpp b/libfcgi/fcgio.cpp
index 1111111..2222222 100644
--- a/libfcgi/fcgio.cpp
+++ b/libfcgi/fcgio.cpp
@@ -18,10 +18,6 @@
 // implied; without even the implied warranty of MERCHANTABILITY or
 // FITNESS FOR A PARTICULAR PURPOSE.
 
-#ifdef _WIN32
-#define DLLAPI  __declspec(dllexport)
-#endif
-
 #include <stdio.h>
 #include <limits.h>
 #include "fcgio.h"
diff --git a/libfcgi/os_win32.c b/libfcgi/os_win32.c
index 1111111..2222222 100755
--- a/libfcgi/os_win32.c
+++ b/libfcgi/os_win32.c
@@ -27,8 +27,6 @@
 #include <process.h>
 #include <signal.h>
 
-#define DLLAPI  __declspec(dllexport)
-
 #include "fcgimisc.h"
 #include "fcgios.h"
 
